src/awkward/_errors.py

```diff
@@ -97,8 +93,8 @@ def decorate_exception(self, cls: type[E], exception: E) -> E:
                     + "\n\nSee if this has been reported at https://github.com/scikit-hep/awkward/issues"
                 )
                 new_exception.__cause__ = exception
-            elif issubclass(cls, KeyError):
-                new_exception = AwkwardKeyError(self.format_exception(exception))
+            elif issubclass(cls, builtins.KeyError):
+                new_exception = KeyError(self.format_exception(exception))
```

Comment:
I prefer not to shadow the `KeyError` builtin, as it's not something one usually overrides in a global namespace. Our `AwkwardKeyError` isn't something I anticipated even L3 awkward modules importing, so changing `__name__` didn't both me. With some time having elapsed, I have changed my mind on that, so I'm now in favour of this change.